src/main/scala/com/databricks/labs/overwatch/utils/Tools.scala

```diff
@@ -560,13 +577,20 @@ object Helpers extends SparkSessionWrapper {
    * @return
    */
   def getWorkspaceByDatabase(
-                              etlDB: String,
+//                              etlDB: String,
+                              db: String,
                               organization_id: Option[String] = None,
                               apiUrl: Option[String] = None,
                               successfullOnly: Boolean = true,
                               disableValidations: Boolean = false
                             ): Workspace = {
     // verify database exists
+    val initialCatalog = spark.sessionState.catalogManager.currentCatalog.name()
+    val etlDB = if(db.contains(".")){
+      spark.sessionState.catalogManager.setCurrentCatalog(db.split("\\.").head)
```

Comment:
is this necessary? If user passes in "catalog.schema" wouldn't the spark.table call just read spark.table("catalog.schema.pipeline_report") which would work anyway? Asking, not telling here. Just curious

Reply:
@GeekSheikh  spark.sessionState.catalog.getDatabaseMetadata(dbName) - this function does not work with catalog.dbName , it fails if input is passed in this format. Thats why we have do add this logic here

Reply:
done.